torch/_inductor/ir.py

```diff
@@ -7550,7 +7550,25 @@ def has_side_effects(self) -> bool:
         return get_schema_info(self.op_overload).is_mutable()
 
     def get_inputs_that_alias_output(self) -> Sequence[str]:
-        return self.alias_names
+        assert isinstance(
+            self.op_overload, (torch._ops.OpOverload, torch._ops.HigherOrderOperator)
+        ), (
+            f"Fails to create FallbackKernel for {self.op_overload}: "
+            f"{type(self.op_overload)} not supported"
+        )
+
+        # See [Note: FallbackKernel supported operators]: for a mutating
```

Comment:
Shouldn't we just update this so as we do not create incorrect alias_names in the first place ?

Reply:
`self.alias_names` are arguments that are aliased which are not not wrong.

The issue is that `def get_inputs_that_alias_output(self) -> Sequence[str]` treats all `inputs arguments that are aliased` as `inputs arguments that alias output`. This is wrong for `mutating op that is auto-functionalizable`.